aptos-move/move-examples/nfts_as_accounts/sources/token_store.move

```diff
@@ -0,0 +1,41 @@
+/// TokenStore holds opaque references to Tokens via TokenRef. An application should be able to
+/// convert any type (container) of TokenRef to a base TokenRef and then store it here. An
+/// application can also remove the TokenRef and convert it back into the appropriate type
+/// (container).
+///
+/// @TODO(davidiw): add transfer, borrow semantics, and other useful helpers
+module nfts_as_accounts::token_store {
+    use std::signer;
+    use aptos_std::table;
+
+    use nfts_as_accounts::token;
+
+    struct TokenStore has key {
+        inner: table::Table<address, token::TokenRef<token::BaseToken>>,
```

Comment:
If that is the only purpose of `BaseToken` I would consider introduce a different type `AbstractTokenRef` and then add the conversions, this seems to be kind of strange to use `TokenRef` for this.

Reply:
Yeah, that's right.